src/hooks/useCredentialApi.ts

```diff
@@ -1,44 +1,170 @@
+ *   - `deleteCredentials(credential)`: Function to handle the entire credential deletion process (API call, response handling, and alerts).
  */
-import React from 'react';
-import axios from 'axios';
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { AlertProps } from '@patternfly/react-core';
+import axios, { AxiosError, AxiosResponse, isAxiosError } from 'axios';
+import { helpers } from '../helpers';
 import { CredentialType } from '../types/types';
 
-const useCredentialApi = () => {
-  const [pendingDeleteCredential, setPendingDeleteCredential] = React.useState<CredentialType>();
-
-  /**
-   * Executes a DELETE request to delete a credential.
-   *
-   * @param {CredentialType} [credential] - (Optional) The credential to be deleted. If not provided, it uses `pendingDeleteCredential`.
-   * @returns {AxiosResponse} - The Axios response object representing the result of the request.
-   */
-  const deleteCredential = (credential?: CredentialType) => {
-    return axios.delete(
-      `${process.env.REACT_APP_CREDENTIALS_SERVICE}${(credential || pendingDeleteCredential)?.id}/`
-    );
-  };
+type ApiDeleteCredentialSuccessType = {
+  message: string;
+  deleted?: number[];
+  missing?: number[];
+  skipped?: { credential: number; sources: number[] }[];
+};
 
-  /**
-   * Deletes several selected credentials based on user interaction.
-   * Add your logic to handle the deletion of selected items within this function.
-   */
-  const onDeleteSelectedCredentials = () => {
-    const selectedItems = [];
-    console.log('Deleting selected credentials:', selectedItems);
-  };
+type ApiDeleteCredentialErrorType = {
+  detail?: string;
+  message: string;
+};
+
+const useDeleteCredentialApi = (onAddAlert: (alert: AlertProps) => void) => {
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (ids: CredentialType['id'][]): Promise<AxiosResponse<ApiDeleteCredentialSuccessType>> =>
+      axios.post(`${process.env.REACT_APP_CREDENTIALS_SERVICE_BULK_DELETE}`, { ids }),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (
+      response: AxiosResponse<ApiDeleteCredentialSuccessType>,
+      updatedCredentials: CredentialType[]
+    ) => {
+      const { data } = response;
+
+      const deletedNames = data?.deleted
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const skippedNames = data?.skipped
+        ?.map(({ credential }) => updatedCredentials.find(cred => credential === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingNames = data?.missing
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingCredsMsg = 'The following credentials could not be found:' + missingNames;
```

Comment:
required: let's be consistent and start avoiding old skool string concats, flip to js template/string literal
```
`The following ... : ${missingNames}`
```